test/dynamo/test_structured_trace.py

```diff
@@ -1120,6 +1108,80 @@ def user_context() -> str:
             f(torch.randn(i + 2 // 3, 5))
             step.next_step()
 
+    def _setup_collective_schedule_capture(self):
+        """Helper to turn on and capture the 'inductor_collective_schedule' structured trace."""
+        payload_buffer = io.StringIO()
+        payload_handler = logging.StreamHandler(payload_buffer)
+        payload_handler.setLevel(logging.DEBUG)
+        payload_handler.setFormatter(StructuredTracePayloadFormatter())
+        payload_handler.addFilter(
+            StructuredTraceTestingFilter("inductor_collective_schedule")
+        )
+        trace_log.addHandler(payload_handler)
+        return payload_buffer, payload_handler
+
+    @requires_tlparse
+    def test_collective_schedule_empty(self):
+        """Verify logging when no collective kernels are present (empty schedule)."""
+        payload_buffer, payload_handler = self._setup_collective_schedule_capture()
+
+        from torch._inductor.debug import log_collective_schedule
+
+        log_collective_schedule([])
+
+        self.assertIn('"inductor_collective_schedule"', self.buffer.getvalue())
+        self.assertEqual(json.loads(payload_buffer.getvalue()), [])
+        self.assertParses()
+
+        trace_log.removeHandler(payload_handler)
```

Comment:
it's better to write _setup_collective_schedule_capture() as a context manager and wrap the test inside the context, instead of manually removing the handler at the end